mne/preprocessing/ica.py

```diff
@@ -1529,16 +1547,27 @@ def apply(self, inst, include=None, exclude=None, n_pca_components=None,
               start=None, stop=None, verbose=None):
         """Remove selected components from the signal.
 
-        Given the unmixing matrix, transform data,
-        zero out components, and inverse transform the data.
+        Given the unmixing matrix, transform the data,
+        zero out all excluded components, and inverse-transform the data.
         This procedure will reconstruct M/EEG signals from which
         the dynamics described by the excluded components is subtracted.
-        The data is processed in place.
 
         Parameters
         ----------
         inst : instance of Raw, Epochs or Evoked
-            The data to be processed. The instance is modified inplace.
+            The data to be processed (i.e., cleaned). It will be modified
+            inplace.
+
+            .. note:: Applying ICA may introduce a DC shift. If you pass
+                      baseline-corrected `~mne.Epochs` or `~mne.Evoked` data,
+                      the baseline period of the cleaned data may not be of
+                      zero mean anymore. If you require baseline-corrected
+                      data, apply baseline correction again after cleaning
+                      via ICA. A warning will be emitted to remind you of this
+                      fact if you pass baseline-corrected data.
+
+            .. versionchanged:: 0.23
+               Warn if instance was baseline-corrected.
```

Comment:
This is not really a change in the parameter itself but rather the behavior / handling of the parameter, so this is probably better in the `Notes` section.

I would also put the `.. note::` in the notes section.

Same for the above docstring changes